### PR TITLE
Fix compiling the examples

### PR DESCRIPTION
**Pull request by masterleinad**
This should avoid all the deprecated functions. I only tested that it compiles, though, and we still need #2687 to be able to compile at all.

### CODE REVIEW

**Comment by masterleinad** [comment]
For testing this, I rebase on top of #2687 and enabled examples by default. I will drop these commits when we decide this is satisfactory.

**Comment by masterleinad** [comment]
The last compile time error to fix is
```
kokkos-develop/example/multi_fem/NonlinearElement_Cuda.hpp(153): error: no instance of constructor "Kokkos::Impl::CudaParallelLaunch<DriverType, Kokkos::LaunchBounds<0U, 0U>, Kokkos::Impl::Experimental::CudaLaunchMechanism::ConstantMemory>::CudaParallelLaunch [with DriverType=HybridFEM::Nonlinear::ElementComputation<HybridFEM::FEMesh<double, 27U, Kokkos::Cuda>, double>]" matches the argument list
            argument types are: (HybridFEM::Nonlinear::ElementComputation<HybridFEM::FEMesh<double, 27U, Kokkos::Cuda>, double>, const dim3, const dim3, const unsigned int)

kokkos-develop/example/multi_fem/NonlinearElement_Cuda.hpp(153): error: no instance of constructor "Kokkos::Impl::CudaParallelLaunch<DriverType, Kokkos::LaunchBounds<0U, 0U>, Kokkos::Impl::Experimental::CudaLaunchMechanism::ConstantMemory>::CudaParallelLaunch [with DriverType=HybridFEM::Nonlinear::ElementComputation<HybridFEM::FEMesh<double, 27U, Kokkos::Cuda>, double>]" matches the argument list
            argument types are: (HybridFEM::Nonlinear::ElementComputation<HybridFEM::FEMesh<double, 27U, Kokkos::Cuda>, double>, const dim3, const dim3, const unsigned int)

kokkos-develop/core/src/impl/Kokkos_FunctorAnalysis.hpp(119): error: static assertion failed with "Execution Policy and Functor execution space must match"
          detected during:
            instantiation of class "Kokkos::Impl::FunctorAnalysis<PatternInterface, Policy, Functor> [with PatternInterface=Kokkos::Impl::FunctorPatternInterface::REDUCE, Policy=Kokkos::RangePolicy<>, Functor=Explicit::internal_force<double, Kokkos::HostSpace::execution_space>]" 
kokkos-develop/core/src/Kokkos_Parallel_Reduce.hpp(1129): here
            instantiation of "void Kokkos::parallel_reduce(const size_t &, const FunctorType &) [with FunctorType=Explicit::internal_force<double, Kokkos::HostSpace::execution_space>]" 
kokkos-develop/example/multi_fem/ExplicitFunctors.hpp(1001): here
            instantiation of "void Explicit::internal_force<Scalar, DeviceType>::apply(const Explicit::internal_force<Scalar, DeviceType>::Fields &, Scalar, int) [with Scalar=double, DeviceType=Kokkos::HostSpace::execution_space]" 
kokkos-develop/example/multi_fem/Explicit.hpp(247): here
            instantiation of "Explicit::PerformanceData Explicit::run<Scalar,FixtureType>(const FixtureType::FEMeshType &, int, int, int, int, int) [with Scalar=double, FixtureType=BoxMeshFixture<double, Kokkos::HostSpace::execution_space, FixtureElementHex8>]" 
kokkos-develop/example/multi_fem/Explicit.hpp(383): here
            instantiation of "void Explicit::driver<Scalar,Device>(const char *, comm::Machine, int, int, int, int) [with Scalar=double, Device=Kokkos::HostSpace::execution_space]" 
kokkos-develop/example/multi_fem/TestHost.cpp(96): here
```

**Comment by crtrott** [comment]
Examples to be removed: fenl, feint, fixture, global_2_local_ids, grow_array, md_skeleton, multi_fem,sort_array and the common directory.

**Comment by dalg24** [comment]
@masterleinad 5234d1a0df143a87d84fdeaa683b811a73a57fbe is for the release candidate only

**Comment by masterleinad** [comment]
This should be OK now.

File: example/CMakeLists.txt [hunk 1]
```diff
@@ -5,16 +5,8 @@
...
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(feint)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(fenl)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(multi_fem)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(md_skeleton)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(global_2_local_ids)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(grow_array)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(sort_array)
-if(NOT Kokkos_ENABLE_CUDA)
-KOKKOS_ADD_EXAMPLE_DIRECTORIES(tutorial)
+if(NOT KOKKOS_ENABLE_CUDA)
```
Comment (dalg24) [inline example/CMakeLists.txt]: Why did you change the case?

File: example/README [hunk 2]
```diff
@@ -5,12 +5,5 @@ Tpetra.
 
 MANIFEST:
 
-  - common:  Header files used by different examples
-  - feint:   Unstructured finite-element method
-  - fixture: Some other finite-element method example
-  - global_2_local_ids: Example of global-to-local index lookup
-  - grow_array:   Parallel dynamic memory allocation
-  - md_skeleton:  Molecular dynamics
```
Comment (dalg24) [inline example/README]: @crtrott where did they move?  Maybe we can update the comment above.

**Review by dalg24 — COMMENTED** [review]
`example/tutorial/README` is grossly outdated
`example/cmake/Dependencies.cmake` can probably be removed

File: example/CMakeLists.txt (hunk 1 above)
Comment (masterleinad) [reply example/CMakeLists.txt]: I thought we want to use the uppercase version everywhere for consistency.

**Comment by masterleinad** [comment]
> `example/tutorial/README` is grossly outdated
> `example/cmake/Dependencies.cmake` can probably be removed

Even if I don't think the purpose of this pull request is to make sure everything there is up-to-date, I cleaned the comments and removed that dependency file in the last commit.
For the future, I am much in favor of reevaluating everything in that folder and convert all the examples to `CMake` projects.

**Comment by ndellingwood** [comment]
@masterleinad @dalg24 @crtrott changes in this PR will also be cherry-picked to the release-candidate-3.0 branch for snapshot into Trilinos, correct?

**Comment by masterleinad** [comment]
Yes, and we will delete the option from `CMake` there.

**Comment by ndellingwood** [comment]
Thanks @masterleinad !

Cross-referencing the PR #2699

File: example/CMakeLists.txt (hunk 1 above)
Comment (crtrott) [reply example/CMakeLists.txt]: Nope. It is always Kokkos_ALL_CAPS for CMake variables and KOKKOS_ALL_CAPS for macros in code

Comment (masterleinad) [reply example/CMakeLists.txt]: Then we are not consistent either
```
$ grep -rIn "KOKKOS_ENABLE_" cmake/ CMakeLists.txt | grep -v "SET" | grep -v "KOKKOS_ENABLE_OPTION" | grep -v "cmakedefine" | wc -l
32
```
Anyway, reverting here.

**Comment by ndellingwood** [comment]
#2699 was merged into release-candidate-3.0 (and will be in master branch when promotion is complete), but this PR hasn't gone into develop yet. Will this PR be merged into develop before master is merged back into develop when the promotion is complete?

**Review by crtrott — APPROVED** [review]

**Comment by ndellingwood** [comment]
After merge of these changes the Makefile build system now fails to compile with messages like:

```
19:03:56 make[2]: Entering directory `/home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_gcc_530_deprecated_off/TestAll_2020-02-05_18.54.22/gcc/5.3.0/Serial-release/example/fixture'
19:03:56 make[2]: /home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_gcc_530_deprecated_off/kokkos/example/fixture/Makefile: No such file or directory
19:03:56 make[2]: *** No rule to make target `/home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_gcc_530_deprecated_off/kokkos/example/fixture/Makefile'.  Stop.
19:03:56 make[2]: Leaving directory `/home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_gcc_530_deprecated_off/TestAll_2020-02-05_18.54.22/gcc/5.3.0/Serial-release/example/fixture'
19:03:56 make[1]: *** [all] Error 2
19:03:56 make[1]: Leaving directory `/home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_gcc_530_deprecated_off/TestAll_2020-02-05_18.54.22/gcc/5.3.0/Serial-release/example/fixture'
19:03:56 make: *** [build-test] Error 2
19:04:04   Starting job gcc-5.3.0-Serial-hwloc-release
19:06:06   FAILED gcc-5.3.0-OpenMP_Serial-release
```

I didn't file an issue because it's not clear to me if the makefile build system is still supposed to be maintained, even though it is deprecated.
